modules/serverpod_auth/serverpod_auth_server/lib/src/business/email_auth.dart

```diff
@@ -504,20 +538,24 @@ class Emails {
     session.log(' - password is correct, userId: ${entry.userId})',
         level: LogLevel.debug);
 
-    var migratedAuth = await Emails.tryMigrateAuthEntry(
-      password: password,
-      entry: entry,
-    );
-
-    if (migratedAuth != null) {
-      session.log(' - migrating authentication entry', level: LogLevel.debug);
-      try {
-        await EmailAuth.db.updateRow(session, migratedAuth);
-      } catch (e) {
-        session.log(
-          ' - failed to update migrated auth: $e',
-          level: LogLevel.error,
-        );
+    if (AuthConfig.current.passwordHashGenerator ==
+            defaultGeneratePasswordHash ||
+        AuthConfig.current.passwordHashValidator ==
+            defaultValidatePasswordHash) {
+      var migratedAuth = await Emails.tryMigrateAuthEntry(
```

Comment:
Suggestion: add a similar guard to the `migrateLegacyPasswordHashes` method but with a throw instead.

Also add the following documentation to the method: Warning: This migration method is designed for password hashes generated by the framework's default algorithm. Hashes stored with a custom generator or different algorithm may produce unexpected results.